docs(start/overview): add `Search Params` section (#12048)
remix-run/react-router: docs/start/overview.md

## Search Params

> React Router can read and modify the [query string][querystring]. This process is handeled with [`URLSearchParams`][urlsearchparams]. We often get and set the Search Params using `useSearchParams`.

Reviewer: ```suggestion
React Router can read and modify the [query string][querystring]. You can set the search params using `useSearchParams`, which returns [`URLSearchParams`][urlsearchparams] and a setter function.
```

```jsx lines=[2,5,8]
function App() {
  const [searchParams, setSearchParams] = useSearchParams();

  // access a specific param
  const query = searchParams.get("q") || "";

  function handleClick() {
    setSearchParams({ q: "example" });
  }

  return (
    <>
      <p>The value of the search param with key q is: {query}</p>
      <button onClick={handleClick}>Change search params</button>
    </>
  );
}
```

> You can also use `Link` or `useNavigate` to change the Search Params.

Reviewer: ```suggestion
You can also use `Link` or `useNavigate` to change the search params.
```
